### PR TITLE
[RTL] Fix RtlImageDirectoryEntryToData for 32 bit images on x64

### PR DESCRIPTION
**Pull request by tkreuzer**
We can't run 32 bit images on x64 yet, but we should at least be able to map them and retrieve their resources, etc.

### CODE REVIEW

**Review by HBelusca — APPROVED** [review]

**Review by binarymaster — APPROVED** [review]

**Review by ThFabba — APPROVED** [review]

File: sdk/lib/rtl/image.c [hunk 1]
```diff
@@ -289,17 +289,40 @@ RtlImageDirectoryEntryToData(
     if (NtHeader == NULL)
         return NULL;
 
-    if (Directory >= SWAPD(NtHeader->OptionalHeader.NumberOfRvaAndSizes))
-        return NULL;
+#ifdef _WIN64
+    if (NtHeader->OptionalHeader.Magic == IMAGE_NT_OPTIONAL_HDR64_MAGIC)
```
Comment (ThFabba) [inline sdk/lib/rtl/image.c:293]: Actually... if you call the user mode version of this function, does it recognize images that don't match your process's bitness?

Comment (learn-more) [reply sdk/lib/rtl/image.c:293]: https://pastebin.com/v4awm9LE

Test succeeds for me as 32 bit test, so at least the 32 bit version can recognize a 64 bit (in-memory) image

Comment (learn-more) [reply sdk/lib/rtl/image.c:293]: image
2k3

**Review by zefklop — APPROVED** [review]

File: sdk/lib/rtl/image.c (hunk 1 above)
Comment (tkreuzer) [reply sdk/lib/rtl/image.c:293]: @learn-more Your paste is gone. Can you repaste is somewhere or commit it?

Comment (learn-more) [reply sdk/lib/rtl/image.c:293]: Added #3383

**Pull request merged**